File: mobile/lib/core/entities/order_item.dart
import 'entity.dart';
import 'product_variant.dart';

class OrderItem implements Entity {
  late final String title;
  late final ProductVariant productVariant;
  late final int quantity;
  late final double total;

  OrderItem.lazy({
    String? title,
    ProductVariant? productVariant,
    int? quantity,
    double? total,
  }) {
    if (title != null) this.title = title;
    if (productVariant != null) this.productVariant = productVariant;
    if (quantity != null) this.quantity = quantity;
    if (total != null) this.total = total;
  }
}

File: bff/src/main/kotlin/me/lucasbatista/vienna/sdk/entity/OrderItem.kt
package me.lucasbatista.vienna.sdk.entity

class OrderItem(
    title: String? = null,
    productVariant: ProductVariant? = null,
    total: Double? = null,
    quantity: Int? = null,
) {
    lateinit var title: String
    lateinit var productVariant: ProductVariant
    var quantity = 0
    var total = 0.0

    init {
        title?.let { this.title = title }
        productVariant?.let { this.productVariant = productVariant }
        quantity?.let { this.quantity = quantity }
        total?.let { this.total = total }
    }
}